testing/trino-product-tests/src/main/java/io/trino/tests/product/iceberg/TestIcebergSparkCompatibility.java

```diff
@@ -2523,8 +2524,10 @@ public void testRegisterTableWithTableLocation(StorageFormat storageFormat)
         String sparkTableName = sparkTableName(baseTableName);
 
         onSpark().executeQuery(format("CREATE TABLE %s (a INT, b STRING, c BOOLEAN) USING ICEBERG TBLPROPERTIES ('write.format.default' = '%s')", sparkTableName, storageFormat));
+        // Run both inserts on Spark; When an insert on Trino fails and is retried, it creates a second metadata file with same sequence number in the filename,
+        // and register_table would detect this ambiguity and fail.
```

Comment:
isn't this a bug then? Customers would hit this too I guess.

Reply:
I think it's specific test? `onTrino()` retries if the exception matches the following condition:
https://github.com/trinodb/trino/blob/79529a769671996fe13ecab022433f603b7703dd/testing/trino-product-tests/src/main/java/io/trino/tests/product/utils/HadoopTestUtils.java#L42-L50

Reply:
I'm not sure why changing to `onSpark()` resolves the flaky issue. The method also contains the same retry logic.

Reply:
I'm hoping it either won't fail or won't create metadata files with the same version number in the name. I don't know why Trino does this.

If users hit this, they can do the same workaround - call `register_table()` with specifying the metadata file.